drivers/cleanup

```diff
@@ -0,0 +1,137 @@
+                     for -G) (DANGEROUS)
+
+Debug:
+    The --debug parameter enables manipulation of LVHD VDIs for debugging
+    purposes.  ** NEVER USE IT ON A LIVE VM **
+    The following parameters are required:
+    -t --debug <cmd> <cmd> is one of "activate", "deactivate", "inflate",
+                     "deflate".
+    -v --vdi_uuid    VDI UUID
+    """
+    #-d --dry-run     don't actually perform any SR-modifying operations
+    print(output)
+    cleanup.Util.log("(Invalid usage)")
+    sys.exit(1)
+
+##############################################################################
+#
+#  CLI
+#
+def main():
+    action = ""
+    uuid = ""
+    background = False
+    force = False
+    dryRun = False
+    maxAge = 0
+    debug_cmd = ""
+    vdi_uuid = ""
+    shortArgs = "gGc:aqxu:bfdt:v:"
+    longArgs = ["gc", "gc_force", "clean_cache", "abort", "query", "disable",
+            "uuid=", "background", "force", "dry-run", "debug=", "vdi_uuid="]
+
+    try:
+        opts, args = getopt.getopt(sys.argv[1:], shortArgs, longArgs)
+    except getopt.GetoptError:
+        usage()
+    for o, a in opts:
+        if o in ("-g", "--gc"):
+            action = "gc"
+        if o in ("-G", "--gc_force"):
+            action = "gc_force"
+        if o in ("-c", "--clean_cache"):
+            action = "clean_cache"
+            maxAge = int(a)
+        if o in ("-a", "--abort"):
+            action = "abort"
+        if o in ("-q", "--query"):
+            action = "query"
+        if o in ("-x", "--disable"):
+            action = "disable"
+        if o in ("-u", "--uuid"):
+            uuid = a
+        if o in ("-b", "--background"):
+            background = True
+        if o in ("-f", "--force"):
+            force = True
+        if o in ("-d", "--dry-run"):
+            cleanup.Util.log("Dry run mode")
+            dryRun = True
+        if o in ("-t", "--debug"):
+            action = "debug"
+            debug_cmd = a
+        if o in ("-v", "--vdi_uuid"):
+            vdi_uuid = a
+
+    if not action or not uuid:
+        usage()
+    if action == "debug" and not (debug_cmd and vdi_uuid) or \
+            action != "debug" and (debug_cmd or vdi_uuid):
+        usage()
+
+    if action != "query" and action != "debug":
+        print("All output goes to log")
+
+    if action == "gc":
+        cleanup.gc(None, uuid, background, dryRun)
+    elif action == "gc_force":
+        cleanup.gc_force(None, uuid, force, dryRun, True)
+    elif action == "clean_cache":
+        cleanup.cache_cleanup(None, uuid, maxAge)
+    elif action == "abort":
+        cleanup.abort(uuid)
+    elif action == "query":
+        print("Currently running: %s" % cleanup.get_state(uuid))
+    elif action == "disable":
+        cleanup.abort_optional_reenable(uuid)
+    elif action == "debug":
+        cleanup.debug(uuid, debug_cmd, vdi_uuid)
+
+
+if __name__ == '__main__':
+    sys.path.append("/opt/xensource/sm/")
```

Comment:
I don't think you need the path override here?

Reply:
It needs to be present, because ```from sm import cleanup``` causes other imports which are still in /opt/xensource/sm for the moment, as it says in the commit message for that change.

It (and the subsequent import) do need to go up to the top of the file to resolve the pylint complaint.